src/PrestaShopBundle/Bridge/Listener/InitControllerListener.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShopBundle\Bridge\Listener;
+
+use Context;
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
+use PrestaShop\PrestaShop\Core\Localization\Locale\Repository;
+use PrestaShopBundle\Bridge\Controller\ControllerBridgeInterface;
+use PrestaShopBundle\Bridge\Controller\ControllerConfigurationFactory;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\ControllerEvent;
+use Tab;
+use Tools;
+
+/**
+ * Init Controller by configuring something needs in all Controller migrate horizontally
```

Comment:
We should probably explain a bit more what the listener is doing (in its description), since this is the first step to getting rid of `legacyLayout` and the use of `AdminLegacyLayoutController` then we need a clear knowledge of which parts are now handled without it

Like:
- assign controller to context
- init context locale
- create initial orizontal controller configuration

Reply:
Some of these parts might be moved someday in a more generic listener which will also be used by migrated controllers so it's better to know what is here to anticipate this move for later